### DISCUSSION TITLE
training a neural network model from 0 with onnxruntime

### DISCUSSION THREAD

**Discussion by feiyang6688** [microsoft/onnxruntime#12030] [Other Q&A]
After read some onnx runtime guidance and examples, I found that in order to train a model with onnx runtime, I need to use pytorch as the frontend or at least have a pretrained/predefine onnx model. Is there any way to train a model only with onnx runtime?

**Comment by ytaous** [comment]
Is there any reason you can't use the latest ortmodule api instead? You can find quite a few examples here:
https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py
We used to have ort only with C++ api, but it's not active anymore.

**Reply by feiyang6688** [reply]
It seems the ortmodule api needs a parameter which is a defined torch model. Is this possible that not using other machine learning frameworks, defining and training the network with only ONNX runtime?
